tests/test_protocol.py

```diff
@@ -57,15 +71,59 @@ def test_protodantic(proto_path: Path):
-        ("pt:optional[pt:int]", "Int64 | None"),
-        ("pt:dict[pt:str, pt:int]", "dict[str, Int64]"),
-        ("pt:list[pt:union[pt:dict[pt:str, pt:int], pt:list[pt:bytes]]]", "list[dict[str, Int64] | list[bytes]]"),
-        ("pt:optional[pt:dict[pt:union[pt:str, pt:int], pt:list[pt:union[pt:float, pt:bool]]]]", "dict[str | Int64, list[Double | bool]] | None")
+        ("pt:int", "conint(ge=Int32.min(), le=Int32.max())"),
+        ("pt:float", "confloat(ge=Double.min(), le=Double.max())"),
+        ("pt:list[pt:int]", "tuple[conint(ge=Int32.min(), le=Int32.max())]"),
+        ("pt:optional[pt:int]", "conint(ge=Int32.min(), le=Int32.max()) | None"),
+        ("pt:dict[pt:str, pt:int]", "dict[str, conint(ge=Int32.min(), le=Int32.max())]"),
+        ("pt:list[pt:union[pt:dict[pt:str, pt:int], pt:list[pt:bytes]]]", "tuple[dict[str, conint(ge=Int32.min(), le=Int32.max())] | tuple[bytes]]"),
+        ("pt:optional[pt:dict[pt:union[pt:str, pt:int], pt:list[pt:union[pt:float, pt:bool]]]]", "dict[str | conint(ge=Int32.min(), le=Int32.max()), tuple[confloat(ge=Double.min(), le=Double.max()) | bool]] | None"),
     ]
 )
 def test_parse_performative_annotation(annotation: str, expected: str):
     """Test parse_performative_annotation"""
     assert performatives.parse_annotation(annotation) == expected
+
+
+@pytest.mark.parametrize("protocol_spec", [
+    PROTOCOL_FILES["balances.yaml"],
+    # PROTOCOL_FILES["bridge.yaml"],
+    # PROTOCOL_FILES["cross_chain_arbtrage.yaml"],
+    PROTOCOL_FILES["default.yaml"],
+    PROTOCOL_FILES["liquidity_provision.yaml"],
+    PROTOCOL_FILES["markets.yaml"],
+    PROTOCOL_FILES["ohlcv.yaml"],
+    # PROTOCOL_FILES["order_book.yaml"],
+    PROTOCOL_FILES["orders.yaml"],
+    PROTOCOL_FILES["positions.yaml"],
+    PROTOCOL_FILES["spot_asset.yaml"],
+    PROTOCOL_FILES["tickers.yaml"],
+])
+def test_scaffold_protocol(protocol_spec: Path):
+    """Test `adev scaffold protocol` command"""
+
+    protocol = read_protocol_spec(protocol_spec)
+
+    repo_root = protodantic.get_repo_root()
+    packages_dir = repo_root / "packages"
+    if packages_dir.exists():
+        raise Exception("Test assumes no packages directory exists in this repo")
+
+    packages_dir.mkdir(exist_ok=False)
+    tmp_test_agent = repo_root / "tmp_test_agent"
+    original_cwd = os.getcwd()
+    try:
+        subprocess.run(["aea", "create", tmp_test_agent.name], check=True, cwd=repo_root)
+        os.chdir(tmp_test_agent)
+
+        result = subprocess.run(["adev", "-v", "scaffold", "protocol", str(protocol_spec)], check=False, text=True, capture_output=True)
+        if result.returncode != 0:
+            raise ValueError(f"Protocol scaffolding failed: {result.stderr}")
+
+        test_dir = packages_dir / protocol.metadata.author / "protocols" / protocol.metadata.name / "tests"
+        exit_code = pytest.main([test_dir, "-vv", "-s", "--tb=long", "-p", "no:warnings"])
+        assert exit_code == 0
```

Comment:
This tests the scaffolded tests:
- `test_custom_types.py`
- `test_{protocol_name}_dialogues.py`
- `test_{protocol_name}_messages.py`

Requires quite a bit of running time and would be nice to separate into different tests with a singular shared `adev scaffold protocol` invocation